pkg/api/pod/util.go

```diff
@@ -832,6 +832,17 @@ func dropDisabledPodStatusFields(podStatus, oldPodStatus *api.PodStatus, podSpec
 		}
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ResourceHealthStatus) {
+		clear := func(csl []api.ContainerStatus) {
```

Comment:
```suggestion
		setAllocatedResourcesStatusToNil := func(csl []api.ContainerStatus) {
```

And use that instead of clear?

Reading this I got really concerned that we were clearing the podStatus when this feature is turned off. I would change the function name to reflect that we are only clearing allocatedResourceStatus.

Reply:
done